backend/src/routes/api/getUserProfileById.js

```diff
@@ -0,0 +1,22 @@
+import { StatusCodes } from 'http-status-codes';
+import Profile from '../../models/Profile.js';
+
+const userProfileEndpoint = async (router) => {
```

Comment:
Please check Discord thread. 

Basically, we want to make our router easy to extend with middlewares. 
In eg. UserRoutes we want to declare routes and pass the middlewares

eg. 
```js
const UserRoutes = (router) => {
   router.get('/user/:id', middleware1, middleware2, ...);
}
```

your middleware can be moved to a separate function so it will be more readable/may be reused